swift/Tests/Blockchains/SolanaTests.swift

```diff
@@ -0,0 +1,136 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+import XCTest
+import TrustWalletCore
+
+class SolanaTests: XCTestCase {
+
+    func testAddressFromPrivateKey() {
+        let privateKey = PrivateKey(data: Data(Base58.decodeNoCheck(string: "A7psj2GW7ZMdY4E5hJq14KMeYg7HFjULSsWSrTXZLvYr")!))!
+        let publicKey = privateKey.getPublicKeyEd25519()
+        let address = SolanaAddress(publicKey: publicKey)
+        XCTAssertEqual(address.description, "7v91N7iZ9mNicL8WfG6cgSCKyRXydQjLh6UYBWwm6y1Q")
+    }
+
+    func testAddressFromPublicKey() {
+        let publicKey = PublicKey(data: Data(Base58.decodeNoCheck(string: "7v91N7iZ9mNicL8WfG6cgSCKyRXydQjLh6UYBWwm6y1Q")!), type: PublicKeyType.ed25519)!
+        let address = SolanaAddress(publicKey: publicKey)
+        XCTAssertEqual(address.description, "7v91N7iZ9mNicL8WfG6cgSCKyRXydQjLh6UYBWwm6y1Q")
+    }
+
+    func testAddressFromString() {
+        let addressString = "7v91N7iZ9mNicL8WfG6cgSCKyRXydQjLh6UYBWwm6y1Q"
+        let address = SolanaAddress(string: addressString)
+        XCTAssertEqual(address!.description, "7v91N7iZ9mNicL8WfG6cgSCKyRXydQjLh6UYBWwm6y1Q")
+    }
+
+    func testTransferSigner() throws {
+        let transferMessage = SolanaTransfer.with {
+            $0.privateKey = Data(Base58.decodeNoCheck( string: "A7psj2GW7ZMdY4E5hJq14KMeYg7HFjULSsWSrTXZLvYr")!)
```

Comment:
if you reuse `A7psj2GW7ZMdY4E5hJq14KMeYg7HFjULSsWSrTXZLvYr` private key, might worth adding as static 
```
class SolanaTests: XCTestCase {
    let privateKey = Data(Base58.decodeNoCheck( string: "A7psj2GW7ZMdY4E5hJq14KMeYg7HFjULSsWSrTXZLvYr")!)
}
```